miniapp/tsgemm.cpp

```diff
@@ -155,21 +158,18 @@ int hpx_main(::variables_map& vm) {
   init_matrix(a_mat, ::ScalarType(1));
   init_matrix(b_mat, ::ScalarType(2));
 
-  // 1. John's branch                               (mpi_futures)
-  // 2. MPI pool with a single core                 (mpi_pool)
-  // 3. Default pool with high priority executor    (priorities)
-  // 4. Default pool with default priority executor (default)
+  // 1. John's branch
+  // 2. MPI pool with a single core
+  // 3. Default pool with high priority executor for MPI
 #if defined(DLAF_WITH_MPI_FUTURES)
   // This needs remain in scope for all uses of hpx::mpi
   std::string pool_name = "default";
   hpx::mpi::enable_user_polling enable_polling(pool_name);
   ExecutorType mpi_executor(comm_grid.fullCommunicator());
 #else
-  using hpx::threads::thread_priority;
-  std::string pool_name = (ps.setup == "mpi_pool") ? "mpi" : "default";
-  auto priority = (ps.setup == "priorities") ? thread_priority::thread_priority_high
-                                             : thread_priority::thread_priority_default;
-  ExecutorType mpi_executor(pool_name, priority);
+  using hpx::threads::thread_priority::thread_priority_high;
+  auto pool_name = (hpx::resource::pool_exists("mpi")) ? "mpi" : "default";  // New API v1.5
```

Comment:
I may have already commented about this, I think it is worth updating the dependency in the `CMakeLists.txt`.